datasource/context.go

```diff
@@ -332,3 +334,46 @@ func (n *NestedContextReader) Row() map[string]value.Value {
 func (n *NestedContextReader) Ts() time.Time {
 	return n.ts
 }
+
+// NewNestedContextReader provides a context reader which prefixes all keys with a name space.  This is useful if you have overlapping
+// field names between ContextReaders within a NestedContextReader.
+func NewNamespacedContextReader(basereader expr.ContextReader, namespace string) expr.ContextReader {
+	if namespace == "" {
+		return basereader //we could also return an error or allow '.' field prefix, but meh...
+	}
+	prefix := namespace + "."
+	return &NamespacedContextReader{basereader, prefix}
+}
+
+type NamespacedContextReader struct {
+	basereader expr.ContextReader
+	prefix     string
+}
+
+func (n *NamespacedContextReader) Get(key string) (value.Value, bool) {
+	if !strings.Contains(key, n.prefix) {
```

Comment:
actually, why don't we use left/right ?

https://github.com/araddon/qlbridge/blob/master/expr/stringutil_test.go#L21